.github/workflows/linux-build-and-test.yaml

```diff
@@ -0,0 +1,143 @@
+name: Regression Linux
+on:
+  schedule:
+    # run daily 20:00 on master branch
+    - cron: '0 20 * * *'
+  push:
+    branches:
+      - prerelease_test
+  pull_request:
+jobs:
+  regress_linux:
+    name: PG${{ matrix.pg }} [${{ matrix.type }}]
+    runs-on: ubuntu-18.04
+    strategy:
+
+# This is the main workflow for linux regression tests
+# Since GitHub Actions don't currently allow dynamic
+# matrix entries, the matrix has all versions but
+# for pull request we only run the configurations
+# where on_pullrequest is true
+
+      matrix:
+        pg: [ "11.0", "11.8", "12.0", "12.3" ]
+        type: [ Debug, Release, ApacheOnly ]
+
+        include:
+          - type: Debug
+            on_pullrequest: true
+            build_type: Debug
+            pg_build_args: --enable-debug --enable-cassert
+            tsdb_build_args: -DCODECOVERAGE=ON
+            coverage: true
+          - type: Release
+            build_type: Release
+          - type: ApacheOnly
+            build_type: Release
+            tsdb_build_args: -DAPACHE_ONLY
+          - pg: "11.0"
+            on_pullrequest: false
+          - pg: "12.0"
+            on_pullrequest: false
+        exclude:
+          - pg: "11.0"
+            type: ApacheOnly
+          - pg: "11.0"
+            type: Release
+          - pg: "12.0"
+            type: ApacheOnly
+          - pg: "12.0"
+            type: Release
+    env:
+      PG_SRC_DIR: /home/runner/postgresql
+    steps:
+
+    - name: Install Dependencies
+      if: matrix.on_pullrequest || github.event_name != 'pull_request'
+      run: sudo apt install lcov
+
+    - name: Cache PostgreSQL
+      if: matrix.on_pullrequest || github.event_name != 'pull_request'
+      id: cache-postgresql
+      uses: actions/cache@v2
+      with:
+        path: ${{ env.PG_SRC_DIR }}
```

Comment:
Is this correct? Why are we caching (only) the source dir rather than the install dir that has the binaries? This makes the cache a lot bigger since you are also caching all the build artifacts (objects files, etc.). Then you also need to repeat the install step in every build. This seems to somewhat negate the purpose of the cache.

If you need both the binaries and the source, then I would create a directory in the workdir with the following structure:

```
mkdir $WORKDIR/postgresql
mkdir $WORKDIR/postgresql/source
mkdir $WORKDIR/postgresql/pgsql
```
Then unpack the source in `source`, configure with `--prefix=$WORKDIR/postgresql`, `make install`, then run `make clean` and cache `$WORKDIR/postgresql`

Reply:
This is intentional, make install is very fast and if we cache the actual place it installs to then we would have to clean it up after every run because timescaledb installs files in there as well and we wouldnt have a clean slate.

Reply:
I see your point about TimescaleDB install files being added to the cache. But you could get around that by copying the install dir and caching only the copy.

I do worry about having a much bigger cache that will slow things down, but maybe it is not a huge issue? Still, there seems to be easy steps to get around the "clean slate" issue by using a copy so why not keep it as efficient as possible?

In any case, please add comments describing why your're doing this because it is non-obvious and I am sure the question will come up again (or someone will change the config without realizing the functionality).